src/main/java/org/springframework/data/solr/core/SolrTemplate.java

```diff
@@ -89,8 +90,9 @@
  * @author Joachim Uhrlass
  * @author Francisco Spaeth
  * @author Shiradwade Sateesh Krishna
+ * @author Venil Noronha
  */
-public class SolrTemplate implements SolrOperations, InitializingBean, ApplicationContextAware {
+public class SolrTemplate implements SolrOperations, MulticoreSolrOperations, InitializingBean, ApplicationContextAware {

```

Comment:
The original problem here was that `SolrTemplate` was querying the set solr core when performing `queryForFacetPage` even when the parameter was a `class` annotated with `SolrDocument` (having a different core name) and being wired with `MulticoreSolrFactory`. So, the problem may not just be solved by 'finding' the appropriate solr instance external to `SolrTemplate`.